src/pybullet_blocks/envs/cleanup_table_env.py

```diff
@@ -0,0 +1,768 @@
+            if np.isclose(node[0], 0):  # Robot
+                assert robot_state is None
+                vec = node[: RobotState.get_dimension()]
+                robot_state = RobotState.from_vec(vec)
+            elif np.isclose(node[0], 2):  # Bin
+                bin_vec = node[1 : BlockState.get_dimension()]
+                bin_state = BlockState.from_vec(bin_vec)
+            elif np.isclose(node[0], 1):  # Toy (LetteredBlockState)
+                vec = node[: LetteredBlockState.get_dimension()]
+                toy_state = LetteredBlockState.from_vec(vec)
+                toy_states.append(toy_state)
+
+        assert robot_state is not None
+        assert bin_state is not None
+
+        toy_states.sort(key=lambda x: x.letter)
+
+        return cls(toy_states, bin_state, robot_state)
+
+
+class ObjaverseLoader:
+    """Helper class for loading Objaverse objects."""
+
+    def __init__(self, config: ObjaverseConfig) -> None:
+        self.config = config
+        os.makedirs(config.cache_dir, exist_ok=True)
+        self._downloaded_objects: dict[str, str] = {}
+        self._setup_ssl_context()
+        self._download_all_objects()
+
+    def get_object_half_extents(
+        self, physics_client_id: int, body_id: int
+    ) -> tuple[float, float, float]:
+        """Get actual half-extents of a loaded object."""
+        aabb_min, aabb_max = p.getAABB(body_id, physicsClientId=physics_client_id)
+        half_extents = tuple((aabb_max[i] - aabb_min[i]) / 2 for i in range(3))
+        return half_extents
+
+    def _setup_ssl_context(self) -> None:
+        """Set up SSL context to handle certificate verification."""
+        ssl_context = ssl.create_default_context()
+        ssl_context.check_hostname = False
+        ssl_context.verify_mode = ssl.CERT_NONE
+        ssl._create_default_https_context = (  # pylint: disable=protected-access
+            lambda: ssl_context
+        )
+
+    def _download_all_objects(self) -> None:
+        """Download all objects defined in configuration."""
+        uids = [obj_config["uid"] for obj_config in self.config.toy_objects.values()]
+        downloaded_objects = objaverse.load_objects(
+            uids=uids,
+            download_processes=1,
+        )
+        self._downloaded_objects = downloaded_objects
+
+    def _convert_glb_to_obj(self, glb_path: str) -> str:
+        """Convert GLB file to OBJ for PyBullet compatibility."""
+        obj_path = glb_path.replace(".glb", ".obj")
+        if os.path.exists(obj_path):
+            return obj_path
+
+        # Load GLB and export to OBJ
+        mesh = trimesh.load(glb_path)
+        if hasattr(mesh, "geometry"):
+            # if it's a Scene, extract the first geometry
+            if len(mesh.geometry) > 0:
+                mesh = list(mesh.geometry.values())[0]
+            else:
+                raise ValueError("No geometry found in GLB file")
+        mesh.export(obj_path)  # type: ignore[no-untyped-call]
```

Comment:
if this trimesh library is untyped, should we add it to the type ignores in `pyproject.toml` to avoid having to individually annotate things?